zephyr/include/rtos/wait.h

```diff
@@ -12,32 +12,31 @@
 #include <stdbool.h>
 #include <zephyr/kernel.h>
 
-/* TODO: use equivalent Zephyr */
 static inline void idelay(int n)
 {
-	while (n--)
-		asm volatile("nop");
+	k_busy_wait((uint32_t)n);
```

Comment:
Hmm, Zephyr docs say the argument is in usecs: "" * This routine causes the current thread to execute a "do nothing" loop for  usec_to_wait microseconds.""

So this would increase the delay quite a bit in practise.

Reply:
+1 to completely different timing introduced by this change on Intel platforms

Btw, in order to follow the process of switching to Zephyr APIs, calls to `idelay()` should be replaced with `k_busy_wait()` and the latter should be implemented in the _lib_ using the old `idelay()` logic for SOF+XTOS configuration.

Reply:
fwiw, wont idelay() depend on clock speed ? i.e. the Zephyr API should take the clock speed into account where as today we dont ? 

We do need to replace idelay with k_busy_wait(), but need to align the current Intel code based on the 400M clock